pypsa/optimization/optimize.py

```diff
@@ -0,0 +1,570 @@
+        solver. It allows the user to
+        add/change constraints and add/change the objective function.
+    **kwargs:
+        Keyword argument used by `linopy.Model.solve`, such as `solver_name`,
+        `problem_fn` or solver options directly passed to the solver.
+
+    Returns
+    -------
+    None.
+    """
+
+    sns = _as_snapshots(n, snapshots)
+    n._multi_invest = int(multi_investment_periods)
+
+    n.consistency_check()
+    m = create_model(n, sns, multi_investment_periods, **model_kwargs)
+    if extra_functionality:
+        extra_functionality(n, sns)
+    kwargs.setdefault("solver_name", "glpk")
+    status, condition = m.solve(**kwargs)
+
+    if status == "ok":
+        assign_solution(n)
+        assign_duals(n)
+        post_processing(n)
+
+    return status, condition
+
+
+def is_documented_by(original):
+    def wrapper(target):
+        target.__doc__ = original.__doc__
+        return target
+
+    return wrapper
+
+
+class OptimizationAccessor:
+    """
+    Optimization accessor for building and solving models using linopy.
+    """
+
+    def __init__(self, network):
+        self._parent = network
+
+    def __call__(self, *args, **kwargs):
+        return optimize(self._parent, *args, **kwargs)
+
+    __call__.__doc__ = optimize.__doc__
+
+    @is_documented_by(create_model)
+    def create_model(self, **kwargs):
+        return create_model(self._parent, **kwargs)
+
+    def solve_model(self, **kwargs):
+        """
+        Solve an already created model and assign its solution to the network.
+
+        Parameters
+        ----------
+        **kwargs:
+            Keyword argument used by `linopy.Model.solve`, such as `solver_name`,
+            `problem_fn` or solver options directly passed to the solver.
+        """
+        n = self._parent
+        m = n.model
+        kwargs.setdefault("solver_name", "glpk")
+        status, condition = m.solve(**kwargs)
+
+        if status == "ok":
+            assign_solution(n)
+            assign_duals(n)
+            post_processing(n)
+
+        return status, condition
+
+    @is_documented_by(assign_solution)
+    def assign_solution(self, **kwargs):
+        return assign_solution(self._parent, **kwargs)
+
+    @is_documented_by(assign_duals)
+    def assign_duals(self, **kwargs):
+        return assign_duals(self._parent, **kwargs)
+
+    @is_documented_by(post_processing)
+    def post_processing(self, **kwargs):
+        return post_processing(self._parent, **kwargs)
+
+    @is_documented_by(iterative_transmission_capacity_expansion)
+    def iterative_transmission_capacity_expansion(self, *args, **kwargs):
+        iterative_transmission_capacity_expansion(self._parent, *args, **kwargs)
+
+    @is_documented_by(security_constraint_optimization)
+    def security_constraint_optimization(self, *args, **kwargs):
+        security_constraint_optimization(self._parent, *args, **kwargs)
+
+    def fix_optimal_capacities(self):
```

Comment:
You could also have a utility function `fix_optimal_dispatch()` that prepares a solved network for the run of `n.pf()`:

```py
n.links_t.p_set = n.links_t.p0
n.generators_t.p_set = n.generators_t.p
n.storage_units_t.p_set = n.storage_units_t.p
...
```

Reply:
love that!